src/core/core.scale.js

```diff
@@ -322,39 +407,37 @@ module.exports = Element.extend({
 	},
 	calculateTickRotation: function() {
 		var me = this;
-		var context = me.ctx;
-		var tickOpts = me.options.ticks;
-		var labels = labelsFromTicks(me._ticks);
-
-		// Get the width of each grid by calculating the difference
-		// between x offsets between 0 and 1.
-		var tickFont = helpers.options._parseFont(tickOpts);
-		context.font = tickFont.string;
+		var options = me.options;
+		var tickOpts = options.ticks;
+		var ticks = me.getTicks();
+		var labelSizes, maxLabelWidth, maxLabelHeight, maxWidth, tickWidth, maxHeight, maxLabelDiagonal;
 
 		var labelRotation = tickOpts.minRotation || 0;
 
-		if (labels.length && me.options.display && me.isHorizontal()) {
-			var originalLabelWidth = helpers.longestText(context, tickFont.string, labels, me.longestTextCache);
-			var labelWidth = originalLabelWidth;
-			var cosRotation, sinRotation;
-
-			// Allow 3 pixels x2 padding either side for label readability
-			var tickWidth = me.getPixelForTick(1) - me.getPixelForTick(0) - 6;
-
-			// Max label rotation can be set or default to 90 - also act as a loop counter
-			while (labelWidth > tickWidth && labelRotation < tickOpts.maxRotation) {
-				var angleRadians = helpers.toRadians(labelRotation);
-				cosRotation = Math.cos(angleRadians);
-				sinRotation = Math.sin(angleRadians);
-
-				if (sinRotation * originalLabelWidth > me.maxHeight) {
-					// go back one step
-					labelRotation--;
-					break;
+		if (me._isVisible() && tickOpts.display) {
```

Comment:
As an optimization, we could short-circuit all this logic if `tickOpts.maxRotation` isn't greater than `labelRotation`. Then we would not have to `computeLabelSizes`, etc.

Reply:
`scale._labelSizes` is used in other places, so I would not skip `computeLabelSizes`, but it's good idea to have `labelRotation < tickOpts.maxRotation` in the next line.

Reply:
Done